src/main/java/com/solvd/api/BaseRestAssuredMethod.java

```diff
@@ -2,84 +2,47 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.solvd.enums.HttpMethodTypeEnum;
+import com.solvd.enums.HttpMethod;
 import com.solvd.factory.ObjectMapperFactory;
-import com.solvd.graphql.GraphQlQuery;
 import com.solvd.utils.FileReader;
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
 import io.restassured.response.Response;
 import io.restassured.specification.RequestSpecification;
 
 public abstract class BaseRestAssuredMethod {
-    private static final String BEARER_TOKEN = FileReader.getConfigValue("token");
-    private static final ObjectMapper objectMapper = ObjectMapperFactory.buildNew();
-    public String endpointUrl;
+    protected static final String BEARER_TOKEN = FileReader.getConfigValue("token");
+    protected static final ObjectMapper objectMapper = ObjectMapperFactory.buildNew();
+    protected String endpointUrl;
 
-    public Response execute(HttpMethodTypeEnum methodType, Object requestBody) {
+    protected RequestSpecification prepareRequest(HttpMethod methodType, Object requestBody) {
         RequestSpecification requestSpec = RestAssured.given()
                 .header("Authorization", "Bearer " + BEARER_TOKEN)
                 .log().all();
 
-        if (methodType == HttpMethodTypeEnum.POST || methodType == HttpMethodTypeEnum.PUT || methodType == HttpMethodTypeEnum.PATCH) {
-            String jsonBody;
+        if (requestBody != null &&
+                (methodType == HttpMethod.POST || methodType == HttpMethod.PUT || methodType == HttpMethod.PATCH)) {
```

Comment:
Let's remove HttpMethod methodType from  prepareRequest and use it in sendRequest